### PR TITLE
Remember list last selected

### PR DESCRIPTION
**Pull request by BrianPugh**
While using Groups, this PR makes an `lv_list` remember the last selected button so that upon focus, the last used button is selected. For the majority of applications, I can imagine that this is the desired behavior of the developer. However, this differs from the original behavior. If default behavior is desired, we could set a flag in `ext` that determines whether to use original behavior (select first item in list) or new behavior (select last item used).

I wanted this action in `ext` as opposed to using global variables and `lv_list_set_btn_selected` because that approach requires redundant code for saving in every `lv_action_t`. A different global variable would also have to be specified for each `lv_list` which may not be known at compile time.

### CODE REVIEW

**Comment by BrianPugh** [comment]
hold on; in some rarer instances this is crashing my code. Investigating

**Comment by BrianPugh** [comment]
Fixed the crashes mentioned in my previous comment. It was because I wasn't setting the `last_sel` to `NULL` when creating the list, so when my program began allocating previously deallocated memory, it was pointing to a nonsense address.

File: lv_objx/lv_list.c [hunk 1]
```diff
@@ -712,7 +720,11 @@ static lv_res_t lv_list_signal(lv_obj_t * list, lv_signal_t sign, void * param)
             if(btn != NULL) {
                 lv_action_t rel_action;
                 rel_action = lv_btn_get_action(btn, LV_BTN_ACTION_CLICK);
-                if(rel_action != NULL) rel_action(btn);
+                if(rel_action != NULL) {
+                    lv_list_ext_t * ext = lv_obj_get_ext_attr(list);
+                    ext->last_sel = btn;
```
Comment (kisvegabor) [inline lv_objx/lv_list.c]: Why save the last_sel only if there is a valid action?
IMO `ext->last_sel = btn;` can go before `if(rel_action != NULL) {`.

**Comment by kisvegabor** [comment]
Great! Thank you!
I've only one question, see above in the review.

File: lv_objx/lv_list.c (hunk 1 above)
Comment (BrianPugh) [reply lv_objx/lv_list.c]: You're right, oversight on my part. For my application, I just absolutely knew i wanted to save the position right before the callback was triggered. Moved it per your recommendation in 92cb0ac

**Comment by kisvegabor** [comment]
Thanks, for the update!

**Pull request merged**